src/btree/bt_vrfy.c

```diff
@@ -260,12 +266,18 @@ __wt_verify(WT_SESSION_IMPL *session, const char *cfg[])
      * itself.
      * - the debug flag is set where we do not clear the record's txn IDs. Visibility rules may not
      * work correctly when we do not clear the record's txn IDs.
+     * - we are verifying only the specified checkpoint.
      */
     skip_hs = strcmp(name, WT_METAFILE_URI) == 0 || strcmp(name, WT_HS_URI) == 0 ||
-      F_ISSET(session, WT_SESSION_DEBUG_DO_NOT_CLEAR_TXN_ID);
+      F_ISSET(session, WT_SESSION_DEBUG_DO_NOT_CLEAR_TXN_ID) || cfg_ckpt.len > 0;
 
     /* Loop through the file's checkpoints, verifying each one. */
     WT_CKPT_FOREACH (ckptbase, ckpt) {
+        /* If the config provides a specific checkpoint, skip all other ones */
+        if (cfg_ckpt.len > 0 && !WT_CONFIG_MATCH(ckpt->name, cfg_ckpt))
+            continue;
+
+        check_done = true;
```

Comment:
The variable `check_done` can be misleading. I think `checkpoint_found` could be better as we may fail right after this which means we would have found a matching checkpoint but failed at verifying it.
```suggestion
        /* We have found a checkpoint to verify. */
        check_done = true;
```

Reply:
Done